demos/smart_classroom_demo/main.cpp

```diff
@@ -760,53 +739,49 @@ int main(int argc, char* argv[]) {
 
         int teacher_track_id = -1;
 
-        if (cap.GrabNext()) {
-            cap.Retrieve(frame);
-        } else {
-            slog::err << "Can't read the first frame" << slog::endl;
-            return 1;
+        std::unique_ptr<ImagesCapture> cap = openImagesCapture(FLAGS_i, FLAGS_loop, 0, FLAGS_limit);
+        cv::Mat frame = cap->read();
+        if (!frame.data) {
+            throw std::runtime_error("Can't read an image from the input");
         }
 
-        if (actions_type != TOP_K) {
-            action_detector->enqueue(frame);
-            action_detector->submitRequest();
-            face_detector->enqueue(frame);
-            face_detector->submitRequest();
-        }
-
-        prev_frame = frame.clone();
-
-        bool is_last_frame = false;
-        bool is_monitoring_enabled = false;
-        auto prev_frame_path = cap.GetVideoPath();
+        cv::Size graphSize{static_cast<int>(frame.cols / 4), 60};
+        Presenter presenter(FLAGS_u, frame.rows - graphSize.height - 10, graphSize);
 
         cv::VideoWriter vid_writer;
         if (!FLAGS_out_v.empty()) {
             vid_writer = cv::VideoWriter(FLAGS_out_v, cv::VideoWriter::fourcc('M', 'J', 'P', 'G'),
-                                         cap.GetFPS(), Visualizer::GetOutputSize(frame.size()));
+                                         cap->fps(), Visualizer::GetOutputSize(frame.size()));
```

Comment:
Need to be careful here. What'll this do if the source is an image/directory?

Reply:
Updated